src/main/java/org/apache/sysds/runtime/matrix/data/LibMatrixBincell.java

```diff
@@ -278,25 +285,62 @@ public static void bincellOpInPlaceRight(MatrixBlock m1ret, MatrixBlock m2, Bina
 	 * 
 	 * m2 op m1ret
 	 * 
+	 * 
+	 * Note that sometimes the returned matrix is not inplace, since it might be optimal to return a new Sparse Matrix instead.
+	 * Therefore to be safe use the returned matrix in continued operations
+	 * 
 	 * @param m1ret result matrix updated in place
 	 * @param m2 matrix block the other matrix to take values from
 	 * @param op binary operator the operator that is placed in the middle of m1ret and m2
+	 * @return The result MatrixBlock
 	 */
-	public static void bincellOpInPlaceLeft(MatrixBlock m1ret, MatrixBlock m2, BinaryOperator op) {
-		if(m1ret.isInSparseFormat() || m2.isInSparseFormat())
-			throw new NotImplementedException("Not implemented sparse inplace left binaryOperator for matrixBlocks");
-		
-		final double[] retV = m1ret.getDenseBlockValues();
-		final double[] m2V = m2.getDenseBlockValues();
+	public static MatrixBlock bincellOpInPlaceLeft(MatrixBlock m1ret, MatrixBlock m2, BinaryOperator op) {
+		final int nRows = m1ret.getNumRows();
+		final int nCols = m1ret.getNumColumns();
+		if(m1ret.isInSparseFormat()){
+			// not doing in place, since the m1ret is in sparse format, and m2 might make it dense.
+			// this is not ideal either, but makes it work
+			LOG.warn("Inefficient bincell op in place left, because output is materialized in new matrix");
+			MatrixBlock ret = new MatrixBlock(nRows, nCols, true);
```

Comment:
please double check that compression is the only place calling this to avoid creating implicitly new return blocks although the caller things this is inplace.